Web Inspector: include `integrity` in "Copy as fetch"

dcrousso opened the pull request:
#### 16121838cc440eafe42d55a0b7f1d18f1cf5da8d
<pre>
Web Inspector: include `integrity` in &quot;Copy as fetch&quot;
<a href="https://bugs.webkit.org/show_bug.cgi?id=241217">https://bugs.webkit.org/show_bug.cgi?id=241217</a>
&lt;rdar://problem/94698949&gt;

Reviewed by Patrick Angle.

* Source/JavaScriptCore/inspector/protocol/Network.json:
* Source/WebCore/inspector/agents/InspectorNetworkAgent.cpp:
(WebCore::buildObjectForResourceRequest):
Add an optional `integrity` string to `Network.Response` and pass it to the frontend.

* Source/WebInspectorUI/UserInterface/Controllers/NetworkManager.js:
(WI.NetworkManager.prototype.resourceRequestWillBeSent):
* Source/WebInspectorUI/UserInterface/Models/Resource.js:
(WI.Resource.prototype.get integrity): Added.
(WI.Resource.prototype.updateForRedirectResponse):
(WI.Resource.prototype.generateFetchCode):
Pass along the `integrity` whenever updating a `WI.Resource` with a `Network.Request`.

* LayoutTests/http/tests/inspector/network/copy-as-fetch.html:
* LayoutTests/http/tests/inspector/network/copy-as-fetch-expected.txt:

Canonical link: <a href="https://commits.webkit.org/251819@main">https://commits.webkit.org/251819@main</a>
</pre>

patrickangle commented on Source/WebInspectorUI/UserInterface/Models/Resource.js:
```diff
@@ -721,7 +723,7 @@ WI.Resource = class Resource extends WI.SourceCode
     }
 
-    updateForResponse(url, mimeType, type, responseHeaders, statusCode, statusText, elapsedTime, timingData, source, security)
+    updateForResponse(url, mimeType, type, responseHeaders, statusCode, statusText, integrity, elapsedTime, timingData, source, security)
```
I'm getting closer to saying this should be a single destructured object each time you add something 😅 – definitely something for a separate patch though.

patrickangle commented on Source/JavaScriptCore/inspector/protocol/Network.json:
```diff
@@ -76,6 +76,7 @@
                 { "name": "mimeType", "type": "string", "description": "Resource mimeType as determined by the browser." },
                 { "name": "source", "type": "string", "enum": ["unknown", "network", "memory-cache", "disk-cache", "service-worker", "inspector-override"], "description": "Specifies where the response came from." },
                 { "name": "requestHeaders", "$ref": "Headers", "optional": true, "description": "Refined HTTP request headers that were actually transmitted over the network." },
+                { "name": "integrity", "type": "string", "optional": true },
```
No description?

dcrousso committed:
Web Inspector: include `integrity` in "Copy as fetch"
https://bugs.webkit.org/show_bug.cgi?id=241217
<rdar://problem/94698949>

Reviewed by Patrick Angle.

* Source/JavaScriptCore/inspector/protocol/Network.json:
* Source/WebCore/inspector/agents/InspectorNetworkAgent.cpp:
(WebCore::buildObjectForResourceRequest):
Add an optional `integrity` string to `Network.Response` and pass it to the frontend.

* Source/WebInspectorUI/UserInterface/Controllers/NetworkManager.js:
(WI.NetworkManager.prototype.resourceRequestWillBeSent):
* Source/WebInspectorUI/UserInterface/Models/Resource.js:
(WI.Resource.prototype.get integrity): Added.
(WI.Resource.prototype.updateForRedirectResponse):
(WI.Resource.prototype.generateFetchCode):
Pass along the `integrity` whenever updating a `WI.Resource` with a `Network.Request`.

* LayoutTests/http/tests/inspector/network/copy-as-fetch.html:
* LayoutTests/http/tests/inspector/network/copy-as-fetch-expected.txt:

Canonical link: https://commits.webkit.org/251819@main
